docs/usage.md

```diff
@@ -0,0 +1,236 @@
+
+- `file-upload.js`: a script that uploads a large file, used to track memory usage and network throughput.
+
+
+## Helper scripts
+
+We use a couple of scripts to help us with running the benchmarks and compiling the results:
+
+- [k6bench.sh](/k6bench.sh): this is a wrapper script around `k6 run` that starts k6 and the metric collection processes in the background.
+  It redirects the k6 output to a file, and outputs metric results to stdout in CSV format every 5 seconds by default.
+
+- [k6bench.gnuplot](/k6bench.gnuplot): this is a [gnuplot](http://www.gnuplot.info/) script that generates a graph from the CSV results created by `k6bench.sh`.
+  It doesn't create the prettiest graphs, but it was chosen because it's scriptable, so it can be easily automated.
+
+
+## Dependencies
+
+The benchmarks depend on a few utilities to collect the performance results:
+
+- [`smem`](https://www.selenic.com/smem/) for RAM usage, specifically proportional set size (PSS).
+
+  As it turns out, it's rather tricky to determine the precise physical RAM usage of a single process in Linux. Tools like `ps` and `htop` report the resident set size (RSS), which includes the size of any shared libraries the process may be using, and is not an accurate value of how much physical memory the process is using. The PSS value is more precise, but is more difficult to measure. `smem` is seemingly the only tool that simplifies tracking this, but it's unfortunately written in Python, so its own performance is not great, and it can use a lot of CPU and take some time to generate the result.
+
+- [`jq`](https://stedolan.github.io/jq/) is used to extract the VUs and RPS values from the JSON output returned by the k6 REST API.
+
+- [`curl`](https://curl.se/) to make requests to the k6 REST API.
+
+- GNU `coreutils` like `ps`, `top`, `cat`, `awk`, and Bash itself, of course.
+
+
+## Benchmarking strategy
+
+We will run one test at a time, to avoid overloading the SUT and skewing the results.
+
+We won't run all scripts on all machines, but will use the following strategy:
+
+- From the machine with the lowest specs, i.e. `m5.large`, we will only run the `website.js` script, with a high sleep time between iterations in order to not overwhelm the load generator. The goal of this test is to track overall CPU and memory usage, not reach maximum throughput.
+
+- From the middle-spec'd machine, i.e. `m5.4xlarge`, we will run the `website.js` script with more VUs and a lower sleep time between iterations. This way we can observe the change in resource consumption and throughput by using the same script with an increased load.
+
+- From the machine with the highest specs, i.e. `m5.24xlarge`, we will run the `website.js` script with even more VUs and a very low sleep time between iterations. We will also run the RPS optimized and file upload test scripts, so we can observe the performance of k6 when it's allowed to fully utilize the system resources.
```

Comment:
Nit: scanning this it is a bit hard to find out we will run the other 2 scrips only in highest machine as we do not use their script names.

Reply:
In step 4 of "Running the benchmarks" this is explained in detail, along with the commands to run on each machine. This is just a general overview of the strategy, and not meant to serve as reference.

What do you suggest? Would mentioning the scripts by name make this clearer?